soccer/gameplay/skills/angle_receive.py

```diff
@@ -0,0 +1,259 @@
+import single_robot_behavior
+import behavior
+import robocup
+import constants
+import main
+import math
+import enum
+import time
+import skills._kick
+
+
+## AngleReceive accepts a receive_point as a parameter and gets setup there to catch the ball
+# It transitions to the 'aligned' state once it's there within its error thresholds and is steady
+# Set its 'ball_kicked' property to True to tell it to dynamically update its position based on where
+# the ball is moving and attempt to catch it.
+# It will move to the 'completed' state if it catches the ball, otherwise it will go to 'failed'.
+# Kick is a single_robot_behavior, so no need to import both
+class AngleReceive(skills._kick._Kick):
+
+    ## max difference between where we should be facing and where we are facing (in radians)
+    FaceAngleErrorThreshold = 8 * constants.DegreesToRadians
+
+    ## how much we're allowed to be off in the direction of the pass line
+    PositionYErrorThreshold = 0.06
+
+    ## how much we're allowed to be off side-to-side from the pass line
+    PositionXErrorThreshold = 0.03
+
+    DribbleSpeed = 70
+
+    ## we have to be going slower than this to be considered 'steady'
+    SteadyMaxVel = 0.04
+    SteadyMaxAngleVel = 3   # degrees / second
+
+    ## after this amount of time has elapsed after the kick and we haven't received the ball, we failed :(
+    ReceiveTimeout = 2
+
+    class State(enum.Enum):
+        ## we're aligning with the planned receive point
+        aligning = 1
+
+        ## being in this state signals that we're ready for the kicker to kick
+        aligned = 2
+
+        ## the ball's been kicked and we're adjusting based on where the ball's moving
+        receiving = 3
+
+    def __init__(self):
+        super().__init__()
+
+        self.ball_kicked = False
```

Comment:
Rather than storing ball_kicked as a variable, you can check if the kicker behavior has completed.  See [this transition](https://github.com/robojackets/robocup-software/blob/master/soccer/gameplay/tactics/coordinated_pass.py#L57) in coordinated_pass.